### PR TITLE
Consolidating image pulling logic

### PR DESCRIPTION
**Pull request by ronnielai**
Moving image puller logic into image manager by consolidating 2 pullers into one implementation.

### CODE REVIEW

File: pkg/kubelet/images/image_manager.go [hunk 1]
```diff
@@ -27,25 +27,27 @@ import (
 	"k8s.io/kubernetes/pkg/util/flowcontrol"
 )
 
-// imagePuller pulls the image using Runtime.PullImage().
-// It will check the presence of the image, and report the 'image pulling',
-// 'image pulled' events correspondingly.
-type parallelImagePuller struct {
+type imageManager struct {
```
Comment (Random-Liu) [inline pkg/kubelet/images/image_manager.go]: nit: Some comments are appreciated. :)

**Comment by Random-Liu** [comment]
LGTM with nits.

**Comment by vishh** [comment]
lgtm

File: pkg/kubelet/images/image_manager.go (hunk 1 above)
Comment (ronnielai) [inline pkg/kubelet/images/image_manager.go]: Comments added.

**Comment by ronnielai** [comment]
Applied LGTM after adding comments.

**Comment by ronnielai** [comment]
@k8s-bot test this #26982

**Pull request merged**